File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using System.IO;
using DevExpress.XtraReports.UserDesigner;
using DevExpress.XtraCharts;

namespace WindowsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        XtraReport ActiveReport;
        private void Form1_Load(object sender, EventArgs e) {
            this.xtraUserControl11.xrDesignPanel1.OpenReport(new XtraReport1());
            this.xtraUserControl11.xrDesignPanel1.SetCommandVisibility(ReportCommand.ShowPreviewTab,CommandVisibility.None);
            this.xtraUserControl11.xrDesignPanel1.SetCommandVisibility(ReportCommand.ShowHTMLViewTab,CommandVisibility.None);
            this.xtraUserControl11.xrDesignPanel1.SetCommandVisibility(ReportCommand.ShowDesignerTab, CommandVisibility.None);
            
        }

        private void xtraTabControl1_SelectedPageChanging(object sender, DevExpress.XtraTab.TabPageChangingEventArgs e) {

            
            if(e.PrevPage.Text == "Design") {
                if( this.xtraUserControl11.xrDesignPanel1.Report == null)
                e.Cancel = true;
                else {
                    MemoryStream ms = new MemoryStream();
                    try{
                        this.xtraUserControl11.xrDesignPanel1.Report.SaveLayout(ms);
                        ms.Seek(0, SeekOrigin.Begin);
                        ActiveReport = XtraReport.FromStream(ms, true);
                        ActiveReport.DataSource =   this.xtraUserControl11.xrDesignPanel1.Report.DataSource ;
                        ActiveReport.DataMember = this.xtraUserControl11.xrDesignPanel1.Report.DataMember;
                    }
                    finally {
                        ms.Close();
                    }
                }
            }
            if(e.Page.Text == "Preview") {
                ActiveReport.CreateDocument();
                this.xtraUserControl21.printControl1.PrintingSystem = ActiveReport.PrintingSystem;

            }
            if(e.Page.Text == "HTML View") {
                try
                {
                    ActiveReport.ExportToHtml(Environment.CurrentDirectory +"\\temp.html");
                    xtraUserControl31.webBrowser1.Navigate(Environment.CurrentDirectory + "\\temp.html");
                }
                catch (Exception ex) {
                    MessageBox.Show(ex.Message);
                }
                 
            }
        }

    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports System.IO
Imports DevExpress.XtraReports.UserDesigner
Imports DevExpress.XtraCharts

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private ActiveReport As XtraReport
		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Me.xtraUserControl11.xrDesignPanel1.OpenReport(New XtraReport1())
			Me.xtraUserControl11.xrDesignPanel1.SetCommandVisibility(ReportCommand.ShowPreviewTab,CommandVisibility.None)
			Me.xtraUserControl11.xrDesignPanel1.SetCommandVisibility(ReportCommand.ShowHTMLViewTab,CommandVisibility.None)
			Me.xtraUserControl11.xrDesignPanel1.SetCommandVisibility(ReportCommand.ShowDesignerTab, CommandVisibility.None)

		End Sub

		Private Sub xtraTabControl1_SelectedPageChanging(ByVal sender As Object, ByVal e As DevExpress.XtraTab.TabPageChangingEventArgs) Handles xtraTabControl1.SelectedPageChanging


			If e.PrevPage.Text = "Design" Then
				If Me.xtraUserControl11.xrDesignPanel1.Report Is Nothing Then
				e.Cancel = True
				Else
					Dim ms As New MemoryStream()
					Try
						Me.xtraUserControl11.xrDesignPanel1.Report.SaveLayout(ms)
						ms.Seek(0, SeekOrigin.Begin)
						ActiveReport = XtraReport.FromStream(ms, True)
						ActiveReport.DataSource = Me.xtraUserControl11.xrDesignPanel1.Report.DataSource
						ActiveReport.DataMember = Me.xtraUserControl11.xrDesignPanel1.Report.DataMember
					Finally
						ms.Close()
					End Try
				End If
			End If
			If e.Page.Text = "Preview" Then
				ActiveReport.CreateDocument()
				Me.xtraUserControl21.printControl1.PrintingSystem = ActiveReport.PrintingSystem

			End If
			If e.Page.Text = "HTML View" Then
				Try
					ActiveReport.ExportToHtml(Environment.CurrentDirectory & "\temp.html")
					xtraUserControl31.webBrowser1.Navigate(Environment.CurrentDirectory & "\temp.html")
				Catch ex As Exception
					MessageBox.Show(ex.Message)
				End Try

			End If
		End Sub

	End Class
End Namespace